### PR TITLE
Lightly refactor main pubsub handler

### PR DESCRIPTION
**Pull request by chadwhitacre**
Part of #482, after #512.

I'm pretty sure what I [want to do](https://github.com/getsentry/eng-pipes/issues/482#issuecomment-1614061897) is loop over multiple orgs in this handler (`getsentry`, `codecov`). This sets me up for that.

~This also switches to requiring that `repos` be in `payload`. Can we do that? 🤔~ _Decided against this for now, too controversial._

### CODE REVIEW

File: src/webhooks/pubsub/index.ts [hunk 1]
```diff
@@ -49,36 +49,27 @@ export const pubSubHandler = async (
...
-
-    await triggerStaleBot(repos, octokit, now);
-    tx.finish();
-
-    return;
+  if (func === undefined) {
+    func = async () => {}; // no-op
+    code = 400;
+  } else {
+    repos = payload.repos || DEFAULT_REPOS;
```
Comment (hubertdeng123) [inline src/webhooks/pubsub/index.ts:65]: is repos here currently being used?

Comment (hubertdeng123) [reply src/webhooks/pubsub/index.ts:65]: ah yes it is

```
await func(repos, octokit, now);
```

File: src/webhooks/pubsub/index.ts [hunk 2]
```diff
@@ -49,36 +49,27 @@
...
+    octokit = await getClient(ClientType.App, GETSENTRY_ORG);
+    now = moment().utc();
   }
 
-  reply.code(400);
-  reply.send();
-
+  reply.code(code);
+  reply.send(); // Respond early to not block the webhook sender
+  await func(repos, octokit, now);
```
[the remark this answers is not in the document]
Comment (hubertdeng123) [reply src/webhooks/pubsub/index.ts:72]: don't think this should be a problem

[the remark this answers is not in the document]
Comment (chadwhitacre) [reply src/webhooks/pubsub/index.ts:72]: Agreed. It's true that `payload.name` is a user-provided value, but we only use it to call `Map.get` on a map that we define. The worst case is that the value isn't present, which we explicitly account for (`undefined`) on the next line. 

https://github.com/getsentry/eng-pipes/blob/78b702140e21b3ed551a301179eafd765924fc40/src/webhooks/pubsub/index.ts#L56-L64

**Review by hubertdeng123 — APPROVED** [review]

**Comment by hubertdeng123** [comment]
> This also switches to requiring that repos be in payload. Can we do that? 🤔 Decided against this for now, too controversial.

What do you mean here? Isn't repos already in the payload?

**Comment by hubertdeng123** [comment]
Do you mean hard requiring repos to be in the payload? And raising if it isn't?

**Comment by chadwhitacre** [comment]
> What do you mean here? Isn't repos already in the payload?

It's optional, though ...

https://github.com/getsentry/eng-pipes/blob/78b702140e21b3ed551a301179eafd765924fc40/src/webhooks/pubsub/index.ts#L17

... and we fall back to a default.

https://github.com/getsentry/eng-pipes/blob/78b702140e21b3ed551a301179eafd765924fc40/src/webhooks/pubsub/index.ts#L65

Where is the payload defined? In GCP somewhere?

**Comment by chadwhitacre** [comment]
> that's defined in the ops repo already so we have full control over that. https://github.com/getsentry/ops/blob/master/terraform/super-big-data/super-big-consumers/schedule.tf

The missing link! Thank you! 😁 🙏

**Pull request merged**